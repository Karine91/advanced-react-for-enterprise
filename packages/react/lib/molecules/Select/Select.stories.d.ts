/// <reference types="react" />
import "@gkarine/scss/lib/Select.css";
declare const _default: {
    title: string;
    decorators: import("@storybook/csf").DecoratorFunction<import("@storybook/csf").AnyFramework>[];
};
export default _default;
export declare const Common: () => JSX.Element;
export declare const RenderOption: () => JSX.Element;
export declare const CustomLabel: () => JSX.Element;
