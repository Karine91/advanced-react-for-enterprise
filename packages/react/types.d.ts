declare module '@ds.e/scss/src/foundation/_variables.scss';