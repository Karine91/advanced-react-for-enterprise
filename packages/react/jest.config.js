module.exports = {
  roots: ["<rootDir>/src"],
  testRegex: "(/.*\\.test)\\.(ts|tsx)$",
  testEnvironment: "jsdom",
  testEnvironmentOptions: {
    url: "http://localhost/",
  },
  setupFilesAfterEnv: ["<rootDir>/jest.setup.ts"],
};
