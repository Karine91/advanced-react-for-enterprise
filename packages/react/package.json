{
  "name": "@gkarine/react",
  "version": "0.1.4",
  "main": "lib/index.js",
  "license": "MIT",
  "scripts": {
    "build": "rollup -c",
    "dev": "yarn build --watch",
    "test": "jest --verbose",
    "test:watch": "yarn test --watch",
    "start-storybook": "start-storybook",
    "build-storybook": "build-storybook",
    "chromatic": "npx chromatic --project-token=b2e82606b07e"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.21.0",
    "@storybook/addon-a11y": "^6.5.16",
    "@storybook/addon-knobs": "^6.4.0",
    "@storybook/addon-storysource": "^6.5.16",
    "@storybook/react": "^6.5.16",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "14",
    "@types/jest": "^29.5.0",
    "@types/react": "^18.0.28",
    "chromatic": "^6.17.2",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0",
    "react": "^18.2.0",
    "rollup": "2.22.1",
    "rollup-plugin-typescript2": "0.27.1",
    "typescript": "^5.0.2"
  },
  "readme": "ERROR: No README data found!",
  "_id": "@gkarine/react@1.0.0",
  "publishConfig": {
    "access": "public"
  },
  "gitHead": "7f82a19a6e59ad98c36e4e0496035a058ec477d4",
  "files": [
    "lib"
  ]
}
