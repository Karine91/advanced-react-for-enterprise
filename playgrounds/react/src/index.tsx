import React from 'react'
import ReactDOM from 'react-dom/client';

import { Button } from '@ds.e/react';
import "@ds.e/scss/lib/Button.css"

const root = ReactDOM.createRoot(document.getElementById('root')!);
root.render(<Button label='Example Button' />)