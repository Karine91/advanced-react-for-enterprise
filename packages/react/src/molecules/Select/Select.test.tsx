import React from "react";
import Select, { RenderOptionProps } from "./Select";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

const options = [
  {
    label: "Strict Black",
    value: "strict-black",
  },
  {
    label: "Heavenly Green",
    value: "heavenly-green",
  },
  {
    label: "Sweet Pink",
    value: "pink",
  },
];

test("renders all options passed to it", () => {
  render(<Select options={options} />);

  expect(screen.getAllByRole("menuitemradio")).toHaveLength(options.length);
});

test("renders options using custom renderOption method if passed as prop", () => {
  const renderOption = ({
    option,
    getOptionRecommendedProps,
  }: RenderOptionProps) => {
    return (
      <li {...getOptionRecommendedProps()}>
        Custom render option {option.label}
      </li>
    );
  };
  render(<Select options={options} renderOption={renderOption} />);

  expect(screen.getAllByRole("menuitemradio")[0]).toHaveTextContent(
    /Custom render option/
  );
});

test("calls the onOptionSelected prop with the selected option and its index if passed", async () => {
  const user = userEvent.setup();

  const handler = jest.fn();
  render(<Select options={options} onOptionSelected={handler} />);

  const button = screen.getByRole("button");
  await user.click(button);
  const option = screen.getByRole("menuitemradio", { name: "Heavenly Green" });
  await user.click(option);

  expect(handler).toHaveBeenCalledWith(options[1], 1);
});

test("the button label changes to the selected option label", async () => {
  const user = userEvent.setup();

  render(<Select options={options} />);

  const button = screen.getByRole("button");
  await user.click(button);
  const option = screen.getByRole("menuitemradio", { name: "Heavenly Green" });
  await user.click(option);

  expect(button).toHaveTextContent(/Heavenly Green/);
});

test("snapshot of the selected option state", async () => {
  const user = userEvent.setup();

  const { asFragment } = render(<Select options={options} />);

  const button = screen.getByRole("button");
  await user.click(button);
  const option = screen.getByRole("menuitemradio", { name: "Heavenly Green" });
  await user.click(option);

  expect(asFragment()).toMatchSnapshot();
});

test("can customize select label", () => {
  render(<Select label="custom label" />);

  const button = screen.getByRole("button");
  expect(button).toHaveTextContent("custom label");
});
