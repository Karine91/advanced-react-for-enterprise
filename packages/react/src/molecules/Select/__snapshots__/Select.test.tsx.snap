// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of the selected option state 1`] = `
<DocumentFragment>
  <div
    class="dse-select"
  >
    <button
      aria-controls="dse-select-list"
      aria-haspopup="true"
      class="dse-select__label"
      data-testid="DseSelectButton"
    >
      <p
        class="dse-text dse-text-base"
      >
        Heavenly Green
      </p>
      <svg
        class="dse-select__caret dse-select__caret--closed"
        fill="none"
        height="1rem"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1rem"
      >
        <path
          d="M19 9l-7 7-7-7"
        />
      </svg>
    </button>
    <ul
      aria-hidden="false"
      class="dse-select__overlay "
      id="dse-select-list"
      role="menu"
      style="top: 10px;"
    >
      <li
        aria-label="Strict Black"
        class="dse-select__option
                                
                                
                            "
        role="menuitemradio"
        tabindex="0"
      >
        <p
          class="dse-text dse-text-base"
        >
          Strict Black
        </p>
      </li>
      <li
        aria-checked="true"
        aria-label="Heavenly Green"
        class="dse-select__option
                                dse-select__option--selected
                                dse-select__option--highlighted
                            "
        role="menuitemradio"
        tabindex="-1"
      >
        <p
          class="dse-text dse-text-base"
        >
          Heavenly Green
        </p>
        <svg
          fill="none"
          height="1rem"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1rem"
        >
          <path
            d="M5 13l4 4L19 7"
          />
        </svg>
      </li>
      <li
        aria-label="Sweet Pink"
        class="dse-select__option
                                
                                
                            "
        role="menuitemradio"
        tabindex="0"
      >
        <p
          class="dse-text dse-text-base"
        >
          Sweet Pink
        </p>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
